### ISSUE TITLE
Allow user to pass custom folds (GroupKFold)

### ISSUE THREAD

**Issue by MarcoGorelli** [vecxoz/vecstack#27]
As far as I can tell, in the current implementation, you can only pass in the number of folds. What if the user wants to pass in a custom folds object (e.g. `sklearn.model_selection.GroupKFold`)?

If this is of interest, I can submit a pull request.

**Comment by vecxoz** [comment]
Hi! Thanks for your contribution.

Unfortunately I don’t plan to add custom cross-validation in the near future. The reasons include some logical and technical limitations of stacking procedure. Please see #21 for more info.

`GroupKFold` is different from what was mentioned in #21 and possibly will be added as a separate option. But it still requires some research so don’t expect to see it in the package soon.

**Comment by MarcoGorelli** [comment]
Hi! That's fine, thanks for your rapid response!

**Comment by MarcoGorelli** [comment]
On a second thought, couldn't we just add a 'groups' parameter, and then
internally create a PredefinedSplit object which we assign to kf_?

This would solve the problem of you not wanting to allow people to assign
the same observation to multiple folds.

I can modify my PR and show you if you're interested.

Either way, thanks for your great package, I really appreciate it!

**Comment by vecxoz** [comment]
Thank you! I'm glad you enjoy the package!

Your thoughts are absolutely rational. But still I want to avoid *group split* in any form at least at current point. The reason is that such split may affect underlying statistical properties. As I said some research have to be conducted to find out if *group split* is appropriate for stacking at all. 

For example let's look at `Variant B` of stacking procedure (you can see picture in [tutorial]( https://github.com/vecxoz/vecstack/blob/master/examples/00_stacking_concept_pictures_code.ipynb)). In order to predict *train set* (out-of-fold) model will be trained on folds of *train set* and during training in given fold will NOT see all groups (this is the purpose of *group split*). But in order to predict *test set* it will be trained on full *train set* and will see all groups together. Depending on sizes of groups and other factors such behavior may introduce substantial difference in distributions between *train set* predictions (out-of-fold) and *test set* predictions which may not be good for models on the next stacking level.

**Comment by MarcoGorelli** [comment]
Ok, that makes sense - thanks for taking time to explain that to me, feel
free to close!
